cmd/podman/common/specgen.go

```diff
@@ -128,18 +130,21 @@ func getIOLimits(s *specgen.SpecGenerator, c *ContainerCLIOpts, args []string) (
 
 func getPidsLimits(s *specgen.SpecGenerator, c *ContainerCLIOpts, args []string) (*specs.LinuxPids, error) {
 	pids := &specs.LinuxPids{}
-	hasLimits := false
-	if c.CGroupsMode == "disabled" && c.PIDsLimit > 0 {
+	if c.CGroupsMode == "disabled" && c.PIDsLimit != 0 {
 		return nil, nil
 	}
+	if c.PIDsLimit < 0 {
+		if rootless.IsRootless() && containerConfig.Engine.CgroupManager != config.SystemdCgroupsManager {
+			return nil, nil
+		}
+		pids.Limit = containerConfig.PidsLimit()
+		return pids, nil
+	}
 	if c.PIDsLimit > 0 {
 		pids.Limit = c.PIDsLimit
-		hasLimits = true
-	}
-	if !hasLimits {
-		return nil, nil
+		return pids, nil
```

Comment:
if `c.PidsLimit == 0` is falling through and returning `nil, nil` the right thing?

Reply:
yes, `PidsLimit==0` means no limits